### ISSUE TITLE
not being able to define per site email address

### ISSUE THREAD

**Issue by sircco** [macropin/django-registration#26]
Since it is already possible to use site for email notifications it would be great if it could be possible
to override settings.DEFAULT_FROM_ADDRESS  from function

**Comment by macropin** [comment]
Somewhat related to #54

**Comment by joshblum** [comment]
@sircco does the fix in https://github.com/macropin/django-registration/pull/54 solve your problem?

**Comment by sircco** [comment]
looking at code, i don't think that this does it unless i can  override this line on per case basis.
`  from_email = getattr(settings, 'REGISTRATION_DEFAULT_FROM_EMAIL', settings.DEFAULT_FROM_EMAIL)`

Our app is whitelabel, meaning that more than one domain can use it, register its own customers (and send registration emails)

**Comment by dicato** [comment]
@sircco the use case makes sense 👍 

If you'd like to implement we are happy to review.

**Comment by joshblum** [comment]
strawman could be to allow `REGISTRATION_DEFAULT_FROM_EMAIL` to be a python dotted `path.to.function` which would point a function that take `site` as an argument and does custom logic to return the email? Thoughts?

```
# path/to.py
def function(site):
   return 'example@example.com'
```

**Comment by joshblum** [comment]
(probably cleaner to have a separate settings variable `REGISTRATION_FROM_EMAIL_FUNCTION` but same idea)

**Comment by sircco** [comment]
How about something like this, overriding default if variable not provided
during instantiation.
I didnt look too much at the code recently but here is some dummy code to
explain my idea.

class Registration(Object):
   def __init___(self, mail_from=None, .... ):
       self.mail_from, = mail_from or
settings.REGISTRATION_DEFAULT_FROM_EMAIL

**Comment by dicato** [comment]
I think it's acceptable to use static configuration for this as well. You can use a dictionary to map emails to sites, just as an example.

**Comment by joshblum** [comment]
@sircco would a static configuration meet your use case?

**Comment by sircco** [comment]
It will work for now. If we get so much traffic on our whitelabel site that we can't manage static setting we will think of something and submit code for review.

**Comment by joshblum** [comment]
Great. Do you want to take a first stab at implementing this?

**Comment by joshblum** [comment]
@sircco take a look at #294 and let me know if this address your use case!

**Comment by sircco** [comment]
should work fine!

**Comment by joshblum** [comment]
Awesome! I'll leave this open until @294 is merged in

**Comment by joshblum** [comment]
closed by #294
